pkg/operator/configobservation/apiserver/observe_cors.go

```diff
@@ -0,0 +1,65 @@
+package apiserver
+
+import (
+	"k8s.io/klog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var clusterDefaultCORSALlowedOrigins = []string{
+	`//127\.0\.0\.1(:|$)`,
+	`//localhost(:|$)`,
+}
+
+func ObserveAdditionalCORSAllowedOrigins(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	const corsAllowedOriginsPath = "corsAllowedOrigins"
+
+	listers := genericListers.(configobservation.Listers)
+	errs := []error{}
+	prevObservedConfig := map[string]interface{}{}
+
+	currentCORSAllowedOrigins, _, err := unstructured.NestedStringSlice(existingConfig, corsAllowedOriginsPath)
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
```

Comment:
if this worked fine, shouldn't we use `currentCORSAllowedOrigins` instead of `prevObservedConfig` instead in the error cases below?

Reply:
You're right, returning `existingConfig` seems safer given the error that could appear while trying to enrich the `currentCORSAllowedOrigins` with cluster defaults, I'll modify it as such.